packages/webpack/src/plugin.js

```diff
@@ -248,66 +250,75 @@ class LavaMoatPlugin {
+                  // Sadly, even treeshaking doesn't eliminate that module. It's left there and failing to work when reached by runtime policy enforcement.
+                  // Below is the most reliable way I've found to date to identify ignored modules.
+                  (module.type === JAVASCRIPT_MODULE_TYPE_DYNAMIC &&
+                    // @ts-expect-error BAD TYPES
+                    module.identifierStr?.startsWith('ignored')) ||
                   // @ts-expect-error BAD TYPES
-                  module.identifierStr?.startsWith('ignored')) ||
-                // @ts-expect-error BAD TYPES
-                module.resource === undefined // better to explicitly list it as unenforceable than let it fall through the cracks
-              ) {
-                unenforceableModuleIds.push(moduleId)
-              } else {
-                knownPaths.push({
-                  path: /** @type {any} */ (module).resource,
-                  moduleId,
-                }) // typescript is complaining about the use of `resource` here, but it's actually there.
-              }
+                  module.resource === undefined // better to explicitly list it as unenforceable than let it fall through the cracks
+                ) {
+                  unenforceableModuleIds.push(moduleId)
+                } else {
+                  knownPaths.push({
+                    path: /** @type {any} */ (module).resource,
+                    moduleId,
+                  }) // typescript is complaining about the use of `resource` here, but it's actually there.
+                }
+              })
             })
-          })
-          diag.rawDebug(4, { knownPaths })
-          PROGRESS.report('pathsCollected')
-
-          diag.rawDebug(2, 'writing policy')
-          // use the generated policy to save the user one additional pass
-          // getting the policy also writes all files where necessary
-          const policyToApply = policyGenerator.getPolicy()
-
-          identifierLookup = generateIdentifierLookup({
-            readableResourceIds: options.readableResourceIds,
-            unenforceableModuleIds,
-            paths: knownPaths,
-            policy: policyToApply,
-            canonicalNameMap,
-          })
-
-          if (unenforceableModuleIds.length > 0) {
-            mainCompilationWarnings.push(
-              new WebpackError(
-                `LavaMoatPlugin: the following module ids can't be controlled by policy and must be ignored at runtime: \n  ${unenforceableModuleIds.join()}`
-              )
+            diag.rawDebug(4, { knownPaths })
+            PROGRESS.report('pathsCollected')
+
+            diag.rawDebug(2, 'writing policy')
+            // use the generated policy to save the user one additional pass
+            // getting the policy also writes all files where necessary
+            const policyToApply = policyGenerator.getPolicy()
+
+            assert(
```

Comment:
why use assert all of the sudden? seems like everywhere else under `pkgs/webpack` we just throw Errors

Reply:
It's more succinct and explains the intention better. Some of our errors should be calls to `assert` and I'm hoping to move in that direction.